spec/solver_spec.rb

```diff
@@ -0,0 +1,63 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Factorial Function Unit Testing'
```

Comment:
Great job team 🤝

- [ ] As stated in the project requirements, kindly add tests for; - - `Negative value which raises an exception as an expectation`, and `Special case - when input is 0, the result should equal 1`.